src/main/java/com/metamx/http/client/response/SequenceInputStreamResponseHandler.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2011 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.http.client.response;
+
+import com.google.common.base.Throwables;
+import com.google.common.io.ByteSource;
+import com.metamx.common.logger.Logger;
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBufferInputStream;
+import org.jboss.netty.handler.codec.http.HttpChunk;
+import org.jboss.netty.handler.codec.http.HttpResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.util.Enumeration;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *
+ */
+public class SequenceInputStreamResponseHandler implements HttpResponseHandler<InputStream, InputStream>
+{
+  private static final Logger log = new Logger(SequenceInputStreamResponseHandler.class);
+  private final AtomicLong byteCount = new AtomicLong(0);
+  private final BlockingQueue<InputStream> queue = new LinkedBlockingQueue<>();
+  private final AtomicBoolean done = new AtomicBoolean(false);
+
+  @Override
+  public ClientResponse<InputStream> handleResponse(HttpResponse response)
+  {
+    try {
+      queue.put(new ChannelBufferInputStream(response.getContent()));
+    }
+    catch (InterruptedException e) {
+      log.error(e, "Queue appending interrupted");
+      Thread.currentThread().interrupt();
+      throw Throwables.propagate(e);
+    }
+    byteCount.addAndGet(response.getContent().readableBytes());
+    return ClientResponse.<InputStream>finished(
+        new SequenceInputStream(
+            new Enumeration<InputStream>()
+            {
+              @Override
+              public boolean hasMoreElements()
+              {
+                // Done is always true until the last stream has be put in the queue.
+                // Then the stream should be spouting good InputStreams.
+                synchronized (done) {
+                  return !done.get() || !queue.isEmpty();
+                }
+              }
+
+              @Override
+              public InputStream nextElement()
+              {
+                try {
+                  return queue.take();
+                }
+                catch (InterruptedException e) {
```

Comment:
I had considered that overall, but most of the logs we have in the Druid impl have a lot of stuff solely related to that particular instance. As such, I would expect an implementation to override any method it needs logs for and log ass appropriate before calling the super method.

Since this is an internal handling, I'll add a simple log here in case higher methods do not catch the thrown interrupt.